joomla/joomla-cms
# [4.3][Guided Tours] When the input field has a value, enable the 'Next' button

Pull Request for Issue #40008.

### Summary of Changes

The 'Next' button in a step can never be disabled if the target input field has a value.
Little code refactoring as well.

### Testing Instructions

Run a tour, for instance 'Create a category'.
Once in the title field, the 'Next' button is disabled because the field is required.
Enter a value in the input field. 
The 'Next' button is not enabled.
Go to the next step and go back to the previous step.
The 'Next' button should still be enabled.
Remove the value in the input field.
The 'Next' button should be disabled.

### Actual result BEFORE applying this Pull Request

When returning to the step, the 'Next' button is disabled.
You need to give focus to the input field AND type something for the 'Next' button to be enabled.

### Expected result AFTER applying this Pull Request

When returning to the step, the 'Next' button is enabled because there is a value for the required input field.

### Link to documentations
Please select:
